File: spring-boot/rest-api/src/main/java/no/acntech/sandbox/resource/GreetingResource.java
package no.acntech.sandbox.resource;

import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import no.acntech.sandbox.model.Greeting;

@RequestMapping(path = "/api/greetings")
@RestController
public class GreetingResource {

    @GetMapping
    public ResponseEntity<Greeting> get(@RequestParam(name = "name", defaultValue = "Nobody") String name) {
        return ResponseEntity.ok(new Greeting("Hello " + name + "!"));
    }
}

File: spring-boot-kotlin/rest-api/src/main/kotlin/no/acntech/sandbox/resource/GreetingResource.kt
package no.acntech.sandbox.resource

import no.acntech.sandbox.model.Greeting
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController

@RequestMapping(path = ["greetings"])
@RestController
class GreetingResource {

    fun get(@RequestParam(name = "name", defaultValue = "Nobody") name: String): ResponseEntity<Greeting> {
        return ResponseEntity.ok(Greeting(name))
    }
}